posthog/api/query.py

```diff
@@ -401,3 +408,27 @@ async def event_stream():
                 "Connection": "keep-alive",
             },
         )
+
+
+def is_insight_query(query):
+    insight_kinds = {
+        "TrendsQuery",
+        "FunnelsQuery",
+        "RetentionQuery",
+        "PathsQuery",
+        "StickinessQuery",
+        "LifecycleQuery",
+    }
+    if getattr(query, "kind", None) in insight_kinds:
+        return True
+    if getattr(query, "kind", None) == "HogQLQuery":
+        return True
+    if getattr(query, "kind", None) == "DataTableNode":
+        source = getattr(query, "source", None)
+        if source and getattr(source, "kind", None) in insight_kinds:
+            return True
+    if getattr(query, "kind", None) == "DataVisualizationNode":
+        source = getattr(query, "source", None)
+        if source and getattr(source, "kind", None) in insight_kinds:
+            return True
```

Comment:
I don't think this is necessary, as the frontend already sends the source query. If we still want this, we probably also want to add `InsightVizNode` and add a helper similar to this frontend side one: https://github.com/PostHog/posthog/blob/master/frontend/src/queries/utils.ts#L74-L80